pkg/channel/distributed/common/kafka/sarama/sarama.go

```diff
@@ -45,28 +41,19 @@ func EnableSaramaLogging(enable bool) {
 
 // Load The Sarama & EventingKafka Configuration From The ConfigMap
 // The Provided Context Must Have A Kubernetes Client Associated With It
-func LoadSettings(ctx context.Context, clientId string, kafkaAuthConfig *client.KafkaAuthConfig) (*sarama.Config, *commonconfig.EventingKafkaConfig, error) {
-	if ctx == nil {
-		return nil, nil, fmt.Errorf("attempted to load settings from a nil context")
-	}
-
-	configMap, err := kubeclient.Get(ctx).CoreV1().ConfigMaps(system.Namespace()).Get(ctx, constants.SettingsConfigMapName, metav1.GetOptions{})
-	if err != nil {
-		return nil, nil, err
+func LoadSettings(ctx context.Context, clientId string, configMap map[string]string, kafkaAuthConfig *client.KafkaAuthConfig) (*sarama.Config, *commonconfig.EventingKafkaConfig, error) {
```

Comment:
If we're going to have this configmap "loader" inside the context anyway, would it save some coding complexity (i.e. having to call configmaploader.FromContext(ctx) and the configmapLoader() call itself, including all of the error handling) to just have this LoadSettings function extract the configmaploader function from the context (which is already a parameter here) and call it, instead of calling that in each place that's calling LoadSettings() ?  It would involve fiddling with the tests a bit and I am not 100% sure the result would be "better" but thought I would mention it.

Reply:
Sounds good. 
I have some changes stashed, but I will get back at this before merging the PR.

Reply:
@eric-sap this is done now, thanks for the suggestion.

Reply:
I like it; thanks!

Reply:
I had to revert this later because I needed to load the config outside this function as well.
Specifically, when I needed to compute the configmap hash on controller start.

Reply:
Doh; thanks for the effort!  Sorry for the extra time!